src/ui/public/vislib_vis_type/vislib_renderbot.js

```diff
@@ -46,7 +46,12 @@ module.exports = function VislibRenderbotFactory(Private) {
   VislibRenderbot.prototype.buildChartData = buildChartData;
   VislibRenderbot.prototype.render = function (esResponse) {
     this.chartData = this.buildChartData(esResponse);
-    this.vislibVis.render(this.chartData, this.uiState);
+    // to allow legend to render first (wait for angular digest cycle to complete)
+    return new Promise((resolve, reject) => {
+      setTimeout(() => {
```

Comment:
renderbot is non-angular library so it doesnt use scope ... do you think its a good idea to introduce it ? i kinda like it to be a separate thing.